gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -2143,49 +2145,91 @@ unsigned char Movable::GetNextFace()
 
 // returns whether we made all pending steps (so, false if we must be called again this tick)
 // we can't just do them all here because the caller might have to update searchmap etc
-bool Movable::DoStep(unsigned int walk_speed, ieDword time)
-{
-	if (!path) {
-		return true;
+bool Movable::DoStep(unsigned int walkAnimFrames, ieDword time) {
+	int stepLength = 566;
+	if (core->HasFeature(GF_BREAKABLE_WEAPONS)) { // BG1
```

Comment:
This actually deserves it's one game flag, however until we figure out the animation side of things for non-pst walking speed, this hack can remain.